src/de.rs

```diff
@@ -212,6 +215,20 @@ impl Deserializer {
         }
         None
     }
+
+    fn deserialize_small_signed<'de, V: de::Visitor<'de>>(&self, visitor: V) -> Result<V::Value> {
+        match self.as_safe_integer() {
+            Some(v) => visitor.visit_i32(v as _),
```

Comment:
This is wrong. With `as` you're casting away the whole range between 2^32...2^53 which will lead to acceptance of invalid numbers for the target data type.

Reply:
You are correct. My bad